CHANGELOG.md

```diff
@@ -58,6 +58,7 @@
 * [ENHANCEMENT] Ingester: reduce locked time while matching postings for a label, improving the write latency and compaction speed. #8327
 * [ENHANCEMENT] Ingester: reduce the amount of locks taken during the Head compaction's garbage-collection process, improving the write latency and compaction speed. #8327
 * [ENHANCEMENT] Query-frontend: log the start, end time and matchers for remote read requests to the query stats logs. #8326 #8370 #8373
+* [ENHANCEMENT] Store-gateway: keep index headers that were loaded before the previous shutdown. #8281
```

Comment:
i'd classify this as a bug fix. The index headers were supposed to be kept loaded anyways. And maybe clarify that it fixes the cases where the store-gateway restarts twice in quick succession.